articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+
+```powershell
+Register-AzResourceProvider -ProviderNamespace Microsoft.DataMigration
+```
+
+## Step 3: Create a resource group
+
+An Azure resource group is a logical container into which Azure resources are deployed and managed. Create a resource group before you can create a virtual machine.
+
+Create a resource group by using the [New-AzResourceGroup](/powershell/module/az.resources/new-azresourcegroup) command.
+
+The following example creates a resource group named *myResourceGroup* in the *West US 2* region under the default subscription *mySubscription*.
+
+```powershell
+# Get the details of resource group
+[string] $Location = "westus2"
+[string] $ResourceGroupName = "myResourceGroup"
+
+$resourceGroup = Get-AzResourceGroup -Name $ResourceGroupName
+if (-not($resourceGroup)) {
+    LogMessage -Message "Creating resource group $ResourceGroupName..." -IsProcessing $true
+    $resourceGroup = New-AzResourceGroup -Name $ResourceGroupName -Location $Location
+    LogMessage -Message "Created resource group - $($resourceGroup.ResourceId)."
+}
+else { LogMessage -Message "Resource group $ResourceGroupName exists." }
+```
+
+## Step 4: Create an instance of Azure Database Migration Service
+
+You can create new instance of Azure Database Migration Service by using the [New-AzDataMigrationService](powershell/module/az.datamigration/new-azdatamigrationservice) command. This command expects the following required parameters:
+* *Azure Resource Group name*. You can use [New-AzResourceGroup](/powershell/module/az.resources/new-azresourcegroup) command to create Azure Resource group as previously shown and provide its name as a parameter.
+* *Service name*. String that corresponds to the desired unique service name for Azure Database Migration Service 
+* *Location*. Specifies the location of the service. Specify an Azure data center location, such as West US or Southeast Asia
+* *Sku*. This parameter corresponds to DMS Sku name. The currently supported Sku name are *Standard_1vCore*, *Standard_2vCores*, *Standard_4vCores*, *Premium_4vCores*.
+* *Virtual Subnet Identifier*. You can use [Get-AzVirtualNetworkSubnetConfig](/powershell/module/az.network/get-azvirtualnetworksubnetconfig) command to get the information of a subnet. 
+
+The following script expects that the *myVirtualNetwork* virtual network exists with a subnet named *default* and then creates a Database Migration Service with the name *myDmService* under the resource group created in **Step 3** and in the same region.
+
+```powershell
+# Get a reference to the DMS service - Create if not exists
+[string] $VirtualNetworkName = "myVirtualNetwork"
+[string] $SubnetName = "default"
+[string] $ServiceName = "myDmService"
+
+$dmsServiceResourceId = "/subscriptions/$($global:currentSubscriptionId)/resourceGroups/$ResourceGroupName/providers/Microsoft.DataMigration/services/$ServiceName"
+$dmsService = Get-AzResource -ResourceId $dmsServiceResourceId -ErrorAction SilentlyContinue
```

Comment:
Should you use Get-AzDms cmdlet instead of Get-AzResource?  That way they don't have to format the resource id manually:

$dmsService = Get-AzDms `
    -ResourceGroupName $ResourceGroupName `
    -Name $ServiceName;

Reply:
And then add the -ErrorAction SilentlyContinue option as well.

Reply:
Get-AzDms does not returns the ResourceId for the resource. So for standardize approach i have used Get-AzResource because it is always good to have the ResourceId in the object for future extensions of automation scripts. Hope that is fine?